encoders/firebase-decoders-json/src/json/java/com/google/firebase/decoders/json/JsonDataDecoderBuilderContext.java

```diff
@@ -92,6 +91,20 @@
     return creationCtx;
   }
 
+  private <T> void handleNullValue(FieldRef<T> fieldRef, CreationContextImpl creationCtx)
+      throws IOException {
+    reader.nextNull();
```

Comment:
Since both checks below fail if the value is `null` I think it's safe for you to *only* read it if you are going to use it (line 105)

Reply:
`handleNullValue` will only be called when the next token is null, and the only purpose of `reader.nextNull()` is to consume the null token, so that the reader can continue reading other tokens, therefore, we need to call it. 
And it's always safe to do it,  since the `fiedRef` is always non-null, even if its pointing value is null.

Reply:
Well, technically, if the `fieldRef` fails any of the checks below we throw a runtime exception, which cannot be caught, so the reader is not going to be consumed any further :-). So, if you move it to line 104, it will still work. But, I think you can leave it as is for clarity.

Reply:
got it, good call!